[TextureMapper] Remove SurfaceUpdateInfo

carlosgcampos opened the pull request:
#### 8a9ddac3f43f8b4bd95d3427a86612e216947c63
<pre>
[TextureMapper] Remove SurfaceUpdateInfo
<a href="https://bugs.webkit.org/show_bug.cgi?id=268447">https://bugs.webkit.org/show_bug.cgi?id=268447</a>

Reviewed by Alejandro G. Castro and Nikolas Zimmermann.

Nowadays it only contains the update rect and buffer, so it&apos;s
simpler to just pass those to updateTile(). Also change
NicosiaPaintingEngine paint function to receive a reference
to the buffer and return void since they always return true.

* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.cpp:
(Nicosia::BackingStore::updateTile):
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngine.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineBasic.cpp:
(Nicosia::PaintingEngineBasic::paint):
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineBasic.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineThreaded.cpp:
(Nicosia::PaintingEngineThreaded::paint):
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineThreaded.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
* Source/WebCore/platform/graphics/texmap/coordinated/SurfaceUpdateInfo.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/Tile.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/TiledBackingStoreClient.h:
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.cpp:
(WebKit::updateBackingStore):

Canonical link: <a href="https://commits.webkit.org/273822@main">https://commits.webkit.org/273822@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/757b12c6e928087c0f6280f12d7071f8a6dcc9df

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36796 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15731 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39056 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39437 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32957 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38041 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18232 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12846 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37358 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13273 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32518 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11601 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11606 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32803 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40686 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33372 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33144 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37526 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11890 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9707 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35644 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13549 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12283 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4764 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12790 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Great refactor.

nikolaszimmermann reviewed (approved):
Thumbs up for simplifying Nicosia.

carlosgcampos commented on Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
```diff
@@ -1186,18 +1186,15 @@ void CoordinatedGraphicsLayer::updateContentBuffers()
-            m_nicosia.backingStore->updateTile(tile.tileID(), updateInfo, tileRect);
+            IntRect updateRect(dirtyRect);
+            updateRect.move(-tileRect.x(), -tileRect.y());
+            m_nicosia.backingStore->updateTile(tile.tileID(), updateRect, tileRect, WTFMove(buffer));
```
I've realized we are actually moving the buffer twice now, I'll submit and updated patch.

carlosgcampos committed:
[TextureMapper] Remove SurfaceUpdateInfo
https://bugs.webkit.org/show_bug.cgi?id=268447

Reviewed by Alejandro G. Castro and Nikolas Zimmermann.

Nowadays it only contains the update rect and buffer, so it's
simpler to just pass those to updateTile(). Also change
NicosiaPaintingEngine paint function to receive a reference
to the buffer and return void since they always return true.

* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.cpp:
(Nicosia::BackingStore::updateTile):
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngine.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineBasic.cpp:
(Nicosia::PaintingEngineBasic::paint):
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineBasic.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineThreaded.cpp:
(Nicosia::PaintingEngineThreaded::paint):
* Source/WebCore/platform/graphics/nicosia/NicosiaPaintingEngineThreaded.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
* Source/WebCore/platform/graphics/texmap/coordinated/SurfaceUpdateInfo.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/Tile.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/TiledBackingStoreClient.h:
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.cpp:
(WebKit::updateBackingStore):

Canonical link: https://commits.webkit.org/273822@main
